### PR TITLE
Fix PG12 build breaks and unused_variable warnings

### LINKED ISSUE #44: Build error against PostgreSQL 12
Hi,

pg_auto_failover cannot be built against PostgreSQL 12:

```
+ PG_CONFIG=/usr/pgsql-12/bin/pg_config
+ /usr/bin/make -j3
ipaddr.c: In function 'findHostnameFromLocalIpAddress':
ipaddr.c:548:12: warning: unused variable 'address_len' [-Wunused-variable]
  socklen_t address_len;
            ^~~~~~~~~~~
ipaddr.c:547:26: warning: unused variable 'address' [-Wunused-variable]
  struct sockaddr_storage address;
                          ^~~~~~~
health_check_worker.c: In function 'HealthCheckWorkerLauncherMain':
health_check_worker.c:233:20: error: implicit declaration of function 'AllocSetContextCreateExtended'; did you mean 'AllocSetContextCreateInternal'? [-Werror=implicit-function-declaration]
  launcherContext = AllocSetContextCreateExtended(CurrentMemoryContext,
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                    AllocSetContextCreateInternal
health_check_worker.c:233:18: error: assignment to 'MemoryContext' {aka 'struct MemoryContextData *'} from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
  launcherContext = AllocSetContextCreateExtended(CurrentMemoryContext,
                  ^
health_check_worker.c: In function 'BuildDatabaseList':
health_check_worker.c:363:9: error: implicit declaration of function 'heap_beginscan_catalog'; did you mean 'table_beginscan_catalog'? [-Werror=implicit-function-declaration]
  scan = heap_beginscan_catalog(pgDatabaseRelation, 0, NULL);
         ^~~~~~~~~~~~~~~~~~~~~~
         table_beginscan_catalog
health_check_worker.c:363:7: error: assignment to 'HeapScanDesc' {aka 'struct HeapScanDescData *'} from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
  scan = heap_beginscan_catalog(pgDatabaseRelation, 0, NULL);
       ^
In file included from /usr/pgsql-12/include/server/postgres.h:46,
                 from health_check_worker.c:13:
health_check_worker.c:365:49: error: passing argument 1 of 'heap_getnext' from incompatible pointer type [-Werror=incompatible-pointer-types]
  while (HeapTupleIsValid(dbTuple = heap_getnext(scan, ForwardScanDirection)))
                                                 ^~~~
/usr/pgsql-12/include/server/c.h:626:48: note: in definition of macro 'PointerIsValid'
 #define PointerIsValid(pointer) ((const void*)(pointer) != NULL)
                                                ^~~~~~~
health_check_worker.c:365:9: note: in expansion of macro 'HeapTupleIsValid'
  while (HeapTupleIsValid(dbTuple = heap_getnext(scan, ForwardScanDirection)))
         ^~~~~~~~~~~~~~~~
In file included from health_check_worker.c:21:
/usr/pgsql-12/include/server/access/heapam.h:120:45: note: expected 'TableScanDesc' {aka 'struct TableScanDescData *'} but argument is of type 'HeapScanDesc' {aka 'struct HeapScanDescData *'}
 extern HeapTuple heap_getnext(TableScanDesc scan, ScanDirection direction);
                               ~~~~~~~~~~~~~~^~~~
cli_config.c: In function 'cli_config_check':
cli_config.c:82:6: warning: unused variable 'errors' [-Wunused-variable]
  int errors = 0;
      ^~~~~~
health_check_worker.c:378:19: error: implicit declaration of function 'HeapTupleGetOid'; did you mean 'HeapTupleIsValid'? [-Werror=implicit-function-declaration]
    entry->dboid = HeapTupleGetOid(dbTuple);
                   ^~~~~~~~~~~~~~~
                   HeapTupleIsValid
health_check_worker.c:387:15: error: passing argument 1 of 'heap_endscan' from incompatible pointer type [-Werror=incompatible-pointer-types]
  heap_endscan(scan);
               ^~~~
In file included from health_check_worker.c:21:
/usr/pgsql-12/include/server/access/heapam.h:119:40: note: expected 'TableScanDesc' {aka 'struct TableScanDescData *'} but argument is of type 'HeapScanDesc' {aka 'struct HeapScanDescData *'}
 extern void heap_endscan(TableScanDesc scan);
                          ~~~~~~~~~~~~~~^~~~
health_check_worker.c: In function 'HealthCheckWorkerMain':
health_check_worker.c:452:21: error: assignment to 'MemoryContext' {aka 'struct MemoryContextData *'} from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
  healthCheckContext = AllocSetContextCreateExtended(CurrentMemoryContext,
                     ^
cli_service.c: In function 'cli_monitor_run':
cli_service.c:161:16: warning: unused variable 'existingPgSetup' [-Wunused-variable]
  PostgresSetup existingPgSetup = { 0 };
                ^~~~~~~~~~~~~~~
cli_service.c:155:7: warning: unused variable 'configFilePath' [-Wunused-variable]
  char configFilePath[MAXPGPATH];
       ^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [<builtin>: health_check_worker.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:16: monitor] Error 2
make[1]: *** Waiting for unfinished jobs....
cli_show.c: In function 'cli_show_uri':
cli_show.c:309:10: warning: unused variable 'monitor' [-Wunused-variable]
  Monitor monitor = { 0 };
          ^~~~~~~
cli_create_drop_node.c: In function 'cli_create_monitor':
cli_create_drop_node.c:427:7: warning: unused variable 'ipAddr' [-Wunused-variable]
  char ipAddr[BUFSIZE];
       ^~~~~~
cli_create_drop_node.c:425:16: warning: unused variable 'pgSetup' [-Wunused-variable]
  PostgresSetup pgSetup = config.pgSetup;
                ^~~~~~~
cli_root.c: In function 'root_options':
cli_root.c:114:6: warning: unused variable 'debugCount' [-Wunused-variable]
  int debugCount = 0;
      ^~~~~~~~~~
pgsetup.c: In function 'get_pgpid':
pgsetup.c:403:7: warning: unused variable 'line' [-Wunused-variable]
  char line[BUFSIZE];
       ^~~~
pgsetup.c:402:6: warning: unused variable 'lineno' [-Wunused-variable]
  int lineno;
      ^~~~~~
cli_do_root.c: In function 'keeper_cli_keeper_setup_getopts':
cli_do_root.c:231:23: warning: unused variable 'errors' [-Wunused-variable]
  int c, option_index, errors = 0;
                       ^~~~~~
cli_do_root.c:231:9: warning: unused variable 'option_index' [-Wunused-variable]
  int c, option_index, errors = 0;
         ^~~~~~~~~~~~
cli_do_root.c:231:6: warning: unused variable 'c' [-Wunused-variable]
  int c, option_index, errors = 0;
      ^
monitor_config.c: In function 'monitor_config_set_setting':
monitor_config.c:442:13: warning: unused variable 'option' [-Wunused-variable]
  IniOption *option;
             ^~~~~~
monitor_pg_init.c: In function 'monitor_pg_init':
monitor_pg_init.c:60:7: warning: unused variable 'postgresInstanceExists' [-Wunused-variable]
  bool postgresInstanceExists = pg_setup_pgdata_exists(&pgSetup);
       ^~~~~~~~~~~~~~~~~~~~~~
```

Can you please take a look?

Thanks, Devrim

### PR DESCRIPTION
**Pull request by mtuncer**
Fixes #44

### CODE REVIEW

**Comment by DimCitus** [comment]
From our discussion, here's an example of `#error` implementation for Postgres versions we don't support: https://github.com/dimitri/prefix/commit/29edb97519b52e4960418c79584dfcd0a41e1a5d#diff-79b812cc0e446a629ad70699a33066b0

It boils down to:

``` c
#error "Unknown or unsupported postgresql version"
```

File: src/monitor/version_compat.h [hunk 1]
```diff
@@ -14,6 +14,11 @@
 
 #include "postgres.h"
 
+/* we do not ssupport PG 12 yet*/
```
Comment (DimCitus) [inline src/monitor/version_compat.h]: Typo on “support”.

Also, I think we only support Postgres 10 and 11, so we might as well change the test to check for the supported versions?

Comment (mtuncer) [reply src/monitor/version_compat.h]: are you okay with use of #pragma ?

Comment (mtuncer) [reply src/monitor/version_compat.h]: done (without pragma)

File: src/monitor/version_compat.h [hunk 2]
```diff
@@ -14,7 +14,12 @@
 
 #include "postgres.h"
 
-#if (PG_VERSION_NUM >= 90600 && PG_VERSION_NUM < 110000)
+/* we only support PG10 and PG11, PG 12 is not supported yet*/
+#if (PG_VERSION_NUM < 100000 || PG_VERSION_NUM >= 120000)
+#error "Unknown or unsupported postgresql version"
+#endif
+
+#if (PG_VERSION_NUM < 110000)
```
Comment (DimCitus) [inline src/monitor/version_compat.h]: It would be easier to read now if that was

``` c
#if (PG_VERSION_NUM == 100000)
```

Comment (mtuncer) [reply src/monitor/version_compat.h]: minor version numbers may interfere. I saw PG 11.2 version number as 110002 so exact match might cause issues

Comment (mtuncer) [reply src/monitor/version_compat.h]: perhaps PG_MAJORVERSION but it is a string.

Comment (DimCitus) [reply src/monitor/version_compat.h]: oh, ok then, good point!

**Pull request merged**